source/common/http/matching/inputs.cc

```diff
@@ -2,13 +2,18 @@
 
 #include "envoy/registry/registry.h"
 
+#include "source/common/http/matching/status_code_input.h"
+
 namespace Envoy {
 namespace Http {
 namespace Matching {
 REGISTER_FACTORY(HttpRequestHeadersDataInputFactory, Matcher::DataInputFactory<HttpMatchingData>);
 REGISTER_FACTORY(HttpResponseHeadersDataInputFactory, Matcher::DataInputFactory<HttpMatchingData>);
 REGISTER_FACTORY(HttpRequestTrailersDataInputFactory, Matcher::DataInputFactory<HttpMatchingData>);
 REGISTER_FACTORY(HttpResponseTrailersDataInputFactory, Matcher::DataInputFactory<HttpMatchingData>);
+REGISTER_FACTORY(HttpResponseStatusCodeInputFactory, Matcher::DataInputFactory<HttpMatchingData>);
+REGISTER_FACTORY(HttpResponseStatusCodeClassInputFactory,
+                 Matcher::DataInputFactory<HttpMatchingData>);
```

Comment:
drive by, this is LEGACY_ALWAYSLINK which means it'll be linked into all binaries even if the extensions using it aren't linked in.
Can we have them be in their own files and libraries to avoid that?

Reply:
done.